### PR TITLE
BC Gov Petroleum And Natural Gas Title Credential

### PR DESCRIPTION
**Pull request by PatStLouis**
This is a draft PR looking to align the BC Gov Energy and Mines Digital Trust framework with the traceability vocab to enable data exchanges. We are looking for feedback from both the traceability group on where these credentials would fit and the BC Gov EMDT governance team. 

[Here's the reference documentation](https://bcgov.github.io/bc-vcpedia/credentials/bc-petroleum-and-natural-gas-title/governance.html)

### CODE REVIEW

File: docs/openapi/components/schemas/common/PetroleumAndNaturalGasTitle.yml [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    description: >-
+      A number is used to uniquely identify a title, assigned based on administrative policy by the TRSB. The title number is associated with the Exploration Licence outlined in Section 126, part 14.126 of the PNG Act
+    type: string
+    $linkedData:
+      term: title_number
+      '@id': https://w3id.org/traceability#title_number
+  title_type:
+    title: Title Type
+    description: >-
+      This field specifies the type of title based on the PNG Act. Title types can include the following: 1. Lease (Part 6); 2. Permit (Part 5); 3. Drilling Licence (Part 5.1), and 4. Storage Reservoir Licence (Part 14.130). Each title has only one title type.
```
Comment (TallTed) [inline docs/openapi/components/schemas/common/PetroleumAndNaturalGasTitle.yml:31]:
```suggestion
      This field specifies the type of title based on the PNG Act. Title types include the following: 1. Lease (Part 6); 2. Permit (Part 5); 3. Drilling Licence (Part 5.1), and 4. Storage Reservoir Licence (Part 14.130). Each title has only one title type.
```

File: docs/openapi/components/schemas/common/PetroleumAndNaturalGasTitle.yml [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    type: string
+    $linkedData:
+      term: title_type
+      '@id': https://w3id.org/traceability#title_type
+  issue_date:
+    title: Issue Date
+    description: >-
+      The date when a title is issued, format is YYYY-MM-DD. Section 64 of the PNG Act
```
Comment (TallTed) [inline docs/openapi/components/schemas/common/PetroleumAndNaturalGasTitle.yml:39]:
```suggestion
      The date when a title is issued, formatted as YYYY-MM-DD. Section 64 of the PNG Act
```

File: docs/openapi/components/schemas/common/PetroleumAndNaturalGasTitle.yml [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    type: string
+    $linkedData:
+      term: issue_date
+      '@id': https://w3id.org/traceability#issue_date
+  effective_date:
+    title: Effective Date
+    description: >-
+      The date when the title becomes effective, format YYYY-MM-DD. This date usually follows after an issuance date and signals the start of a title period. This date does not change.
```
Comment (TallTed) [inline docs/openapi/components/schemas/common/PetroleumAndNaturalGasTitle.yml:47]:
```suggestion
      The date when the title becomes effective, formatted as YYYY-MM-DD. This date usually follows an issuance date and signals the start of a title period. This date does not change.
```

File: docs/openapi/components/schemas/common/PetroleumAndNaturalGasTitle.yml [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    type: string
+    $linkedData:
+      term: effective_date
+      '@id': https://w3id.org/traceability#effective_date
+  term:
+    title: Term
+    description: >-
+      This describes the length of time (term) granted for each title, expressed in YYYY. This date does not change.
```
Comment (TallTed) [inline docs/openapi/components/schemas/common/PetroleumAndNaturalGasTitle.yml:55]:
```suggestion
      This describes the length of time (term) granted for each title, formatted as YYYY. This date does not change.
```

File: docs/openapi/components/schemas/common/PetroleumAndNaturalGasTitle.yml [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    type: string
+    $linkedData:
+      term: term
+      '@id': https://w3id.org/traceability#term
+  expiry_date:
+    title: Expiry Date
+    description: >-
+      Expiry date is defined as effective date plus term, expressed as YYYY-MM-DD. This date signals the end of the title period.
```
Comment (TallTed) [inline docs/openapi/components/schemas/common/PetroleumAndNaturalGasTitle.yml:63]:
```suggestion
      Expiry date is defined as effective date plus term, formatted as YYYY-MM-DD. This date signals the end of the title period.
```

File: docs/openapi/components/schemas/common/PetroleumAndNaturalGasTitle.yml [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+    description: >-
+      This attribute is the surface area in planned use, measured in hectares. This includes 3 dimensional layers.
+    type: string
+    $linkedData:
+      term: area
+      '@id': https://w3id.org/traceability#area
+  title_holders:
+    title: Title Holders
+    description: >-
+      The PNG Act describes a "holder of a location" as meaning "in accordance with the context, a permittee, licensee" . In this credential, the term title holder is used to reference a person in whose name a PNG title document is recorded in the division records. Title holders can be registered companies, and/or persons. A Title Holder will hold a percentage (%) interest in the title. Percent interest defines the percentage of interest allotted to each party named in the PNG title document. Percentage of ownership can be divided up to eight digits and should always equal 100% total. Lessee is defined by the PNG Act as "a person in whose name a lease is recorded in the division records".
```
Comment (TallTed) [inline docs/openapi/components/schemas/common/PetroleumAndNaturalGasTitle.yml:79]:
```suggestion
      The PNG Act describes a "holder of a location" as meaning "in accordance with the context, a permittee, licensee". In this credential, the term "title holder" is used to reference a person in whose name a PNG title document is recorded in the division records. Title holders can be registered companies and/or persons. A Title Holder will hold a percentage (%) interest in the title. Percent interest defines the percentage of interest allotted to each party named in the PNG title document. Percentage of ownership can be divided up to eight digits and should always equal 100% total. Lessee is defined by the PNG Act as "a person in whose name a lease is recorded in the division records".
```

**Review by TallTed — CHANGES_REQUESTED** [review]

File: docs/openapi/components/schemas/credentials/PetroleumAndNaturalGasTitleCredential.yml [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+  credentialSubject:
+    type: object
+    properties:
+      type:
+        type: array
+        items:
+          type: string
+          enum:
+            - PetroleumAndNaturalGasTitle
+      title_number:
```
Comment (nissimsan) [inline docs/openapi/components/schemas/credentials/PetroleumAndNaturalGasTitleCredential.yml:66]: snake_casing really burns my eyes... We use camelCase consistently everywhere else.

**Review by nissimsan — CHANGES_REQUESTED** [review]
Please update snake to camel case.

**Comment by PatStLouis** [comment]
@nissimsan These attributes are taken directly from an existing published anoncreds schema. The camelCase/snake_case topic was brought to BC digital team and I'm looking at options such as redefining the schema using camelCase. Until I have agreement from the BC digital team I wouldn't feel comfortable merging this with camelCase. All of the current BC Verifiable Credentials use snake_case unfortunately, we might end up maintaining a separate context in the long run.

**Comment by mkhraisha** [comment]
@PatStLouis has there been updates or movement here

**Comment by nissimsan** [comment]
@PatStLouis , any progress on this?

**Comment by PatStLouis** [comment]
@nissimsan I just received access to the Integrated Petroleum System. This will let me provide test values for an example and update the PR. We can expect an update in the next 2 weeks!

**Comment by PatStLouis** [comment]
Closing this as BC Gov will pursue the UNTP Conformity Credential model for it's use case

**Pull request closed without merging**